public.tf

```diff
@@ -34,7 +34,7 @@ resource "aws_route_table" "public" {
 
   route {
     cidr_block = "0.0.0.0/0"
-    gateway_id = "${var.igw_id}"
```

Comment:
@comeanother there was some sort of race condition doing it this way with the `data.aws_internet_gateway.default.id`, no? I think we had it this way, but then removed it because when creating new VPCs, this would error.

Reply:
If it has issues, then we definitely can provide `igw_id` as a variable.
@comeanother please advise

Reply:
There is a race condition.
When we try to create `VPC` using [terraform-aws-vpc](https://github.com/cloudposse/terraform-aws-vpc) which leverages [terraform-aws-dynamic-subnets](https://github.com/cloudposse/terraform-aws-dynamic-subnets), we get the following error:
![2](https://user-images.githubusercontent.com/11299538/30863520-eb3390b8-a2d9-11e7-8e18-5355efd124b2.png)

`Subnets` are being created at the same time as `IGW`.

In order to get proper sequence, we should add either `depends_on` capability to [terraform-aws-dynamic-subnets](https://github.com/cloudposse/terraform-aws-dynamic-subnets) module or leave `var.igw_id` as a required parameter.

I'd like to keep `var.igw_id` as a required parameter, because it simplifies module structure a lot.

Reply:
ok I see.
This is an issue when using the module with the `terraform-aws-vpc` module.
On its own, this module would be OK.
But I agree, it's not a big deal to just provide the `igw_id` so both modules would work.
I'll delete the PR